### PR TITLE
[1880] Fix OO hex tile reservations

### PR DESCRIPTION
**Pull request by crericha**
The current fix in #8871 assumes all tiles with more than one city is an OO. This is not a valid way to identify an OO city. There are many games with more than one extra city on a tile that is not an OO tile and thus the name of the feature is incorrect.

This fix attempts to provide a clearer blocking mechanism that can be reused without providing unexpected behavior.

Also removed yellow_only as it was added for 1880 and not used by any other game.

### CODE REVIEW

**Comment by michaeljb** [comment]
@crericha sorry about #8871, my mistake.

**Comment by michaeljb** [comment]
@crericha LGTM but I can let someone else approve/merge (even if it's just you 🙂)

**Comment by crericha** [comment]
> @crericha sorry about #8871, my mistake.

No worries. I encourage reviewers to be super critical of common code changes and to think about them from a the general use case. It's super easy for us developers (myself included) to tunnel-vision on the game we are developing and miss the forest from the trees.

> @crericha LGTM but I can let someone else approve/merge (even if it's just you 🙂)

The currently deployed code will work for 1880, so no rush to merge. I'd like Ben to weigh-in and see if he agrees before merging since this PR reworks his PR.

**Review by bentziaxl — APPROVED** [review]

**Comment by bentziaxl** [comment]
Yeah, I definitely had tunnel-vision when fixing this. Thanks @crericha for cleaning my mess.

@michaeljb don't feel bad for approving my fix, it was a clean change, logic was sound, i would have approved it as well.

**Pull request merged**
